@W-19468069 Custom label assessment enhancements

mdmehran-qureshi opened the pull request:
feat: enhance custom label assessment and migration reporting
- Improve custom label assessment with detailed status tracking
- Add comprehensive migration reporting for custom labels
- Update interfaces to support enhanced custom label functionality
- Enhance results builder for better custom label reporting
- Fix TypeScript linting issues in interfaces

mdmehran-qureshi committed:
Merge pull request #2 from mdmehran-qureshi/prerelease/develop-ga

Prerelease/develop ga

mdmehran-qureshi committed:
feat: enhance custom label assessment and migration reporting

- Improve custom label assessment with detailed status tracking
- Add comprehensive migration reporting for custom labels
- Update interfaces to support enhanced custom label functionality
- Enhance results builder for better custom label reporting
- Fix TypeScript linting issues in interfaces

sf-kishore-kurri commented on src/commands/omnistudio/migration/migrate.ts:
```diff
@@ -387,6 +387,7 @@ export default class Migrate extends OmniStudioBaseCommand {
               name: r.name,
               data: this.mergeRecordAndUploadResults(r, cls),
               errors: r.errors,
+              totalCount: r.totalCount, // Preserve totalCount for custom labels
```
@mdmehran-qureshi what is this totalCount variable for? We already display the total objects count in dashboard know

mdmehran-qureshi replied:
In case of custom label, we are not sending all rows to report, Hence sending totalCount explicitly.

sf-kishore-kurri commented on src/commands/omnistudio/migration/migrate.ts:
```diff
@@ -659,38 +660,59 @@ export default class Migrate extends OmniStudioBaseCommand {
-        } else {
-          obj.status = 'Successfully migrated';
+      // For custom labels, preserve all the custom fields
+      if (migrationTool.getName().toLowerCase().includes('custom labels')) {
```
Can you please add the label to Constants and use here

mdmehran-qureshi replied:
Done

sf-kishore-kurri commented on src/commands/omnistudio/migration/migrate.ts:
```diff
@@ -659,38 +660,59 @@ export default class Migrate extends OmniStudioBaseCommand {
+          coreInfo: customLabelRecord.coreInfo,
+          packageInfo: customLabelRecord.packageInfo,
+          message: customLabelRecord.message,
+          cloneStatus: customLabelRecord.cloneStatus,
```
cloneStatus and status will be different in migration?

mdmehran-qureshi replied:
Yes

sf-kishore-kurri commented on src/migration/customLabels.ts:
```diff
@@ -73,63 +90,82 @@ export class CustomLabelsMigrationTool extends BaseMigrationTool implements Migr
+          // Only include error and duplicate (where message is not "same value")
+          if (
+            labelResult.status === 'error' ||
+            (labelResult.status === 'duplicate' && labelResult.message !== 'same value')
```
Can we add list invalid statuses and do the check instead of individual checks

mdmehran-qureshi replied:
Done

sf-kishore-kurri commented on src/utils/resultsbuilder/CustomLabelMigrationReporter.ts:
```diff
@@ -218,66 +226,57 @@ export class CustomLabelMigrationReporter {
+            undefined,
+            statusClass
+          ),
+          createRowDataParam('summary', info.message || '', false, 1, 1, false, undefined, undefined, undefined, true),
```
@mdmehran-qureshi We got one feedback during the GA demo that, only status should display in different colors whereas message should be displayed in black.

mdmehran-qureshi replied:
Already following this

sf-kishore-kurri reviewed (commented):
@mdmehran-qureshi Added comments

snehaljha-sf commented on src/utils/resultsbuilder/CustomLabelMigrationReporter.ts:
```diff
@@ -218,66 +226,57 @@ export class CustomLabelMigrationReporter {
+        rowId: `custom_label_${rowIndex++}`,
+        data: [
+          createRowDataParam('name', info.labelName, true, 1, 1, false),
+          createRowDataParam('packageId', info.packageInfo?.id || '<ID>', false, 1, 1, true),
```
can we use empty string if pakckageInfo.id is mot present to have consistent across other reports

mdmehran-qureshi replied:
Done

mdmehran-qureshi committed:
refactor: replace hardcoded strings with constants for custom label migration

- Add CustomLabelInvalidStatuses, CustomLabelErrorStatus, CustomLabelDuplicateStatus, and CustomLabelSameValueMessage constants
- Replace hardcoded 'custom labels' string with Constants.CustomLabelPluralName in migrate.ts
- Refactor individual status checks to use constants list in customLabels.ts
- Improve code maintainability and consistency with existing patterns

sf-kishore-kurri reviewed (approved):
Changes are looking good to me, but let's merge the changes next week

mdmehran-qureshi committed:
Merge branch 'prerelease/develop-ga' into feature/custom-label-assessment-enhancements
